numpy/_core/src/multiarray/multiarraymodule.c

```diff
@@ -692,7 +692,7 @@ PyArray_ConcatenateInto(PyObject *op,
         Py_DECREF(item);
     }
 
-    if (axis >= NPY_MAXDIMS) {
+    if (axis == NPY_RAVEL_AXIS) {
```

Comment:
Also a subtle oddity, mentioned in the release note (any huge axes was interpreted as raveling)